File: amazon_test.py

import unittest
import os
from selenium import webdriver

class AmazonTest(unittest.TestCase):
    def setUp(self):
        chromedriver = './drivers/chromedriver.exe'
        os.environ['webdriver.chrome.driver'] = chromedriver
        self.driver = webdriver.Chrome(chromedriver)
        self.driver.get('http://www.amazon.com/')

    def tearDown(self):
        self.driver.quit()

    def test_amazon_search(self):
        input_elem = self.driver.find_element_by_id('twotabsearchtextbox')
        input_elem.send_keys("selenium webdriver")
        search_icon = self.driver.find_element_by_css_selector('#nav-search > form > div.nav-right > div > input')
        search_icon.click()

        status = self.driver.find_element_by_id('s-result-count')
        status_text = '1-16 of 139 results for "selenium webdriver"'

        self.assertEqual(status_text, status.text)

if __name__ == '__main__':
    unittest.main()

File: AmazonTest.java

import static org.junit.Assert.assertEquals;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;

public class AmazonTest {
	private WebDriver driver;
	
	@Before
	public void setUp() {
		System.setProperty("webdriver.chrome.driver","./drivers/chromedriver");
		driver = new ChromeDriver();
		driver.get("http://www.amazon.com/");
	}
	
	@After
	public void tearDown() {
		driver.quit();
	}
	
	@Test
	public void testAmazonSearchResult() {
		WebElement input = driver.findElement(By.id("twotabsearchtextbox"));
		input.sendKeys("selenium webdriver");
		WebElement searchIcon =driver.findElement(By.cssSelector("#nav-search > form > div.nav-right > div > input"));
		
		searchIcon.click();
		WebElement status = driver.findElement(By.id("s-result-count"));
		String text = "1-16 of 136 results for \"selenium webdriver\"";
		assertEquals(text, status.getText());
	}
}
